docs/StorageManagement.md

```diff
@@ -0,0 +1,281 @@
+# Storage Management by Fybrik
+
+## Background
+In several use-cases Fybrik needs to allocate storage for data. One use case is implicit copy of a dataset in read scenarios made for performance, cost or governance sake. A second scenario is when a new dataset is created by the workload. In this case Fybrik allocates the storage and registers the new dataset in the data catalog. A third use case is explicit copy - i.e. the user indicates that a copy of an existing dataset should be made. As in the second use case, here too Fybrik allocates storage for the data and registers the new dataset in the data catalog.
+
+When we say that Fybrik allocates storage, we actually mean that Fybrik allocates a portion of an existing storage account for use by the given dataset. Fybrik must be informed what storage accounts are available, and how to access them. This information is currently provided via the FybrikStorageAccount CRD.  
+
+Modules that write data to this storage, receive from Fybrik a connection that holds the relevant information about the storage (e.g. endpoint and write credentials).
+
+Currently, only S3 storage is supported. Both allocation and deletion of the storage (if temporary) is done using Datashim.
+
+## Gaps / Requirements
+
+- Support additional connection types (e.g. MySQL, googlesheets)
+
+- Business logic should not be hard-coded in Fybrik. 
+
+- A single connection taxonomy should be used by modules, catalog conector and storage manager.
+
+- Deployment of FybrikStorageAccount CRD should be configurable - to be discussed: [issue 1717](https://github.com/fybrik/fybrik/issues/1717) 
```

Comment:
If `FybrikStorageAccount` is not deployed, does it mean that the Storage Manager will not been deployed too? Should we support optional deployment of the Storage Manager?

Reply:
Optional deployment of StorageManager is equivalent to optional deployment of FybrikStorageAccount CRD, there is no meaning of deploying one without another. In this design both are deployed unconditionally.